src/util.ts

```diff
@@ -209,3 +210,15 @@ export function varName(s: string): string {
   // Add _ if the string has leading numbers.
   return (s.match(/^\d+/) ? '_' : '') + alphanumericS;
 }
+
+export function logicalExpr<T>(op: LogicalOperand<T>, cb: Function): string {
+  if (isLogicalNot(op)) {
+    return '!' + logicalExpr(op.not, cb);
```

Comment:
Oh I see. In such cases, I wonder whether the corresponding callback function should be responsible for wrapping the string in parentheses? It's seems specific to the case that `op` is a logical expression. It also results in slightly more convoluted strings for selection predicates:

```diff
{"or": ["one", {"and": ["two", {"not": "three"}]}]}
-      '(vlPoint("one_store", "", datum, "union", "all") || ' +
-      '(vlPoint("two_store", "", datum, "union", "all") && ' +
-      '!vlInterval("three_store", "", datum, "intersect", "others")))');
+      '(vlPoint("one_store", "", datum, "union", "all")) || ' +
+      '((vlPoint("two_store", "", datum, "union", "all")) && ' +
+      '(!(vlInterval("three_store", "", datum, "intersect", "others"))))');
```
Anyways, fairly minor point so I've pushed the requested change in 11fa46e.